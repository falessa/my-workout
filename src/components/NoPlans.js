import React from 'react';
import { StyleSheet, View, Text, Button } from 'react-native';

const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: '#fff',
        alignItems: 'center',
        justifyContent: 'center',
    }
})

const NoPlans = ({ onPress }) => {
    return (
        
        <View style={styles.container}>
            <Text>Add a workout plan to start training now!</Text>
            <Button title='Add workout plan' onPress={onPress}/>
        </View>
        
    )
}

export default NoPlans