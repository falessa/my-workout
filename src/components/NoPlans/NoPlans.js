import React from 'react';
import { StyleSheet, View } from 'react-native';
import MainButton from '../MainButton';
import TextPrimary from '../Text/TextPrimary/TextPrimary';
import TextSecondary from '../Text/TextSecondary/TextSecondary'
import WorkoutImage from '../../../assets/svg/personal_training';
import { useTranslation } from 'react-i18next';

const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: '#fff',
        alignItems: 'center',
        justifyContent: 'center',
    }
})

const NoPlans = ({ onPress }) => {
    const { t } = useTranslation()

    return (
        <View>
            <View style={{ alignItems: 'center' }}>
                <WorkoutImage width={320} height={250}/>
            </View>
            <TextPrimary text={t('welcome')}/>
            <TextSecondary text={t('noPlans')} />
            <MainButton onPress={onPress} text={t('addPlan')}/>

        </View>
        
    )
}

export default NoPlans