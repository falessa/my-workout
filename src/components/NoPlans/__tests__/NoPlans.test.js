import "react-native";
import React from "react";
import { create } from "react-test-renderer";
import NoPlans from '../NoPlans';

describe('NoPlans tests', () => {
    const props = {
        onPress: () =>{
            // nothing
        }
    }

    const wrapper = create(<NoPlans {...props} />)

    test('Render NoPlans', () => {
        expect(wrapper).toMatchSnapshot()
    })
})