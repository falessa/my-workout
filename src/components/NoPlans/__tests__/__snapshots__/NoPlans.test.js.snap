// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoPlans tests Render NoPlans 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text>
    noPlans
  </Text>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#386fa4",
        "borderRadius": 20,
        "opacity": 1,
        "paddingVertical": 5,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 18,
          "fontWeight": "bold",
          "margin": 10,
          "textAlign": "center",
        }
      }
    >
      addPlan
    </Text>
  </View>
</View>
`;
